nvflare/apis/stream_shareable.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from builtins import dict as StreamMeta
+from typing import Any, Dict, Tuple
+
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+
+
+class StreamMetaKey:
+    CHANNEL = "__channel__"
+    TOPIC = "__topic__"
+    RC = "__RC__"
+
+
+class StreamShareableGenerator(ABC):
+    @abstractmethod
+    def get_next(
```

Comment:
This is the iterator pattern. We can use Iterator interface and pass the parameters in constructor.

Reply:
It's not necessarily the iterator pattern since it could be called from multiple threads (when/if we implement the streaming with multiple threads) at the same time.

To avoid confusion, will change the class name to ObjectProducer, and the method name to "produce".